File: discarded/tauri-plugin-device-info/android/src/main/java/DeviceInfoPlugin.kt
package com.plugin.device_info

import android.app.Activity
import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkCapabilities
import android.os.BatteryManager
import android.os.Build
import app.tauri.annotation.Command
import app.tauri.annotation.InvokeArg
import app.tauri.annotation.TauriPlugin
import app.tauri.plugin.JSObject
import app.tauri.plugin.Plugin
import app.tauri.plugin.Invoke

@InvokeArg
class PingArgs {
  var value: String? = null
}

@TauriPlugin
class DeviceInfoPlugin(private val activity: Activity): Plugin(activity) {
    
    @Command
    fun ping(invoke: Invoke) {
        val args = invoke.parseArgs(PingArgs::class.java)
        val ret = JSObject()
        ret.put("value", args.value ?: "default value")
        invoke.resolve(ret)
    }
    
    @Command
    fun getDeviceInfo(invoke: Invoke) {
        val ret = JSObject()
        ret.put("osName", "Android")
        ret.put("osVersion", Build.VERSION.RELEASE)
        ret.put("deviceModel", Build.MODEL)
        ret.put("deviceManufacturer", Build.MANUFACTURER)
        invoke.resolve(ret)
    }
    
    @Command
    fun getNetworkInfo(invoke: Invoke) {
        val ret = JSObject()
        
        val connectivityManager = activity.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val network = connectivityManager.activeNetwork
        
        if (network == null) {
            ret.put("isConnected", false)
            ret.put("networkType", "none")
            invoke.resolve(ret)
            return
        }
        
        ret.put("isConnected", true)
        
        val capabilities = connectivityManager.getNetworkCapabilities(network)
        val networkType = when {
            capabilities?.hasTransport(NetworkCapabilities.TRANSPORT_WIFI) == true -> "wifi"
            capabilities?.hasTransport(NetworkCapabilities.TRANSPORT_ETHERNET) == true -> "ethernet"
            capabilities?.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR) == true -> "cellular"
            else -> "unknown"
        }
        
        ret.put("networkType", networkType)
        invoke.resolve(ret)
    }
    
    @Command
    fun getBatteryInfo(invoke: Invoke) {
        val ret = JSObject()
        
        val batteryManager = activity.getSystemService(Context.BATTERY_SERVICE) as BatteryManager
        val batteryLevel = batteryManager.getIntProperty(BatteryManager.BATTERY_PROPERTY_CAPACITY)
        val isCharging = batteryManager.isCharging
        
        ret.put("level", batteryLevel)
        ret.put("isCharging", isCharging)
        
        invoke.resolve(ret)
    }
}

File: discarded/tauri-plugin-device-info/ios/Sources/deviceInfoPlugin.swift
import SwiftRs
import Tauri
import UIKit
import WebKit

class PingArgs: Decodable {
  let value: String?
}

class DeviceInfoPlugin: Plugin {
  @objc public func ping(_ invoke: Invoke) throws {
    let args = try invoke.parseArgs(PingArgs.self)
    invoke.resolve(["value": args.value ?? ""])
  }
  
  @objc public func getDeviceInfo(_ invoke: Invoke) throws {
    invoke.resolve([
      "osName": "iOS",
      "osVersion": UIDevice.current.systemVersion,
      "deviceModel": UIDevice.current.model,
      "deviceManufacturer": "Apple"
    ])
  }
  
  @objc public func getNetworkInfo(_ invoke: Invoke) throws {
    // iOS 端网络信息获取 - 简化版
    // 只判断是否 WiFi 连接，不区分具体网络类型
    let networkStatus = getNetworkStatus()
    invoke.resolve(networkStatus)
  }
  
  @objc public func getBatteryInfo(_ invoke: Invoke) throws {
    // 启用电池监控
    UIDevice.current.isBatteryMonitoringEnabled = true
    
    let level: Int
    if UIDevice.current.batteryLevel == UIDevice.BatteryLevel.unknown {
      level = 100
    } else {
      level = Int(UIDevice.current.batteryLevel * 100)
    }
    
    let isCharging = UIDevice.current.batteryState == .charging || 
                     UIDevice.current.batteryState == .full
    
    invoke.resolve([
      "level": level,
      "isCharging": isCharging
    ])
  }
  
  private func getNetworkStatus() -> [String: Any] {
    // 使用 Network Framework 或 Reachability 检测网络
    // 这里简化实现，实际项目中建议使用 SystemConfiguration
    // 注意：iOS 沙盒限制，无法直接判断 WiFi/蜂窝，只能检测连通性
    return [
      "isConnected": true,
      "networkType": "unknown"
    ]
  }
}

@_cdecl("init_plugin_device_info")
func initPlugin() -> Plugin {
  return DeviceInfoPlugin()
}
